src/ccsn/ccsn.hpp

```diff
@@ -0,0 +1,156 @@
+// Administration. The Government is granted for itself and others
+// acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works,
+// distribute copies to the public, perform publicly and display
+// publicly, and to permit others to do so.
+
+#ifndef CCSN_CCSN_HPP_
+#define CCSN_CCSN_HPP_
+
+// Parthenon
+#include <kokkos_abstraction.hpp>
+#include <parthenon/package.hpp>
+#include <utils/error_checking.hpp>
+
+// Spiner
+#include <spiner/interpolation.hpp>
+
+using namespace parthenon::package::prelude;
+
+namespace CCSN {
+constexpr int NCCSN = 11;
+
+using parthenon::Real;
+using State_t = parthenon::ParArray2D<Real>;
+using State_host_t = typename parthenon::ParArray2D<Real>::HostMirror;
+
+constexpr int R = 0;
+constexpr int RHO = 1;
+constexpr int TEMP = 2;
+constexpr int YE = 3;
+constexpr int EPS = 4;
+constexpr int VEL_RAD = 5;
+constexpr int PRES = 6;
+constexpr int RHO_ADM = 7;
+constexpr int J_ADM = 8;
+constexpr int S_ADM = 9;
+constexpr int Srr_ADM = 10;
+
+using Radius = Spiner::RegularGrid1D;
+
+KOKKOS_INLINE_FUNCTION
+std::pair<int, int> Get1DProfileNumZones(const std::string model_filename) {
+
+  // open file
+  std::ifstream inputfile(model_filename);
+  const std::string whitespace(" \t\n\r");
+
+  // error check
+  if (!inputfile.is_open()) {
+    std::cout << model_filename << " not found :( \n.";
+    exit(-1);
+  }
+
+  int nz = 0;
+  int nc = 0;
+  std::string line;
+  std::string comment1("#");
+  std::string comment2("//");
+
+  // get number of zones from 1d file
+  while (!inputfile.eof()) {
+
+    getline(inputfile, line);
+
+    std::size_t first_nonws = line.find_first_not_of(whitespace);
+
+    // skip empty lines
+    if (first_nonws == std::string::npos) {
+      continue;
+    }
+
+    // skip comments
+    if (line.find(comment1) == first_nonws || line.find(comment2) == first_nonws) {
+      nc++;
+      continue;
+    }
+
+    nz++;
+  }
+
+  inputfile.close();
+
+  return std::make_pair(nz + 1, nc);
+}
+
+template <typename ParArray2D>
+KOKKOS_INLINE_FUNCTION void Get1DProfileData(const std::string model_filename,
+                                             const int num_zones, const int num_comments,
+                                             const int num_vars,
+                                             ParArray2D &ccsn_state_raw_h) {
+
+  std::ifstream inputfile(model_filename);
+
+  Real val = 0;
+  std::string line;
+  int line_num = 0;
+
+  // skip comment lines
+  while (line_num < num_comments) {
+    getline(inputfile, line);
+    line_num++;
+  }
+
+  // read file into model_1d
+  for (int i = 0; i < num_zones; i++) // number of zones
+  {
+    for (int j = 0; j < num_vars; j++) //  number of vars
+    {
+      inputfile >> val;
+      ccsn_state_raw_h(j, i) = val;
+    }
+  }
+
+  std::cout << "Read 1D profile into state array.\n";
+
+  inputfile.close();
+}
```

Comment:
I think there's probably a way to read the data without looping through the file twice. But that's a problem for future us. This is fine for now.